docs/user-guide/concepts/expressions.md

```diff
@@ -21,7 +21,10 @@ Polars performs these core data transformations very quickly by:
 - automatic query optimization on each expression
 - automatic parallelization of expressions on many columns
 
-Polars expressions are a mapping from a series to a series (or mathematically `Fn(Series) -> Series`). As expressions have a `Series` as an input and a `Series` as an output then it is straightforward to do a sequence of expressions (similar to method chaining in pandas).
+Polars expressions are a mapping from a dataframe to a series (or mathematically `Fn(DataFrame) -> Series`). As expressions have a `Series` as output then it is straightforward to do a sequence of expressions (similar to method chaining in pandas) which just keep transforming the output `Series` from the previous step.
```

Comment:
I believe it is `Fn(Series+, Groups?) -> Series`. The context (e.g. `DataFrame`) determine the materialization of one or multiple `Series`. There can be groups that also influence the materialization.

@orlp what was your definition?

Reply:
re-reading the definitive guide definition, that one might be the best one to teach? I've pushed a commit with that one, but keeping the current comment about chaining